src/main/java/com/google/cloud/tools/maven/SdkResolver.java

```diff
@@ -0,0 +1,193 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.maven;
+
+import static com.google.common.collect.Iterables.find;
+
+import com.google.common.io.ByteStreams;
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.io.Files;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+import org.eclipse.aether.resolution.VersionRangeRequest;
+import org.eclipse.aether.resolution.VersionRangeResolutionException;
+import org.eclipse.aether.resolution.VersionRangeResult;
+import org.eclipse.aether.version.Version;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * Resolves the sdk in the .m2 repository, retrieving the appropriate version for this version of
+ * the plugin if necessary.
+ *
+ * @author Matt Stephenson <mattstep@google.com>
+ */
+// TODO(joaomartins): This class needs some rewriting.
+public class SdkResolver {
+
+  private static final String SDK_GROUP_ID = "com.google.appengine";
+  private static final String SDK_ARTIFACT_ID = "appengine-java-sdk";
+  private static final String SDK_EXTENSION = "zip";
+
+  public static File getSdk(MavenProject project, RepositorySystem repoSystem,
+      RepositorySystemSession repoSession, List<RemoteRepository>... repos)
+      throws MojoExecutionException {
+    Artifact artifact = (Artifact) find(project.getPluginArtifacts(), new Predicate<Artifact>() {
+      @Override
+      public boolean apply(Artifact artifact1) {
+        return artifact1.getArtifactId().equals("gcp-app-maven-plugin");
+      }
+    });
+
+    String version = artifact.getVersion();
+
+    if (version.endsWith("-SNAPSHOT")) {
+      String newestVersion = determineNewestVersion(repoSystem, repoSession, repos);
+      return getSdk(newestVersion, repoSystem, repoSession, repos);
+    }
+
+    return getSdk("1.9.32", repoSystem, repoSession, repos);
```

Comment:
@ludoch can you also update the version here when you update it in the gcloud repo?